File: back/src/dtos/roster.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
#[derive(Debug)]
pub struct Roster {
    pub roster_id : i32,
    pub campus_id : i32,
    pub teacher_name : String,
    pub teacher_lastname1 : String,
    pub teacher_lastname2 : String,
    pub subject_name : String,
}

File: back-csharp/_models/Roster.cs
using System;
using System.Collections.Generic;

namespace back_csharp._models
{
    public partial class Roster
    {
        public Roster()
        {
            Comments = new List<Comment>();
        }

        public int RosterId { get; set; }
        public Guid RecordId { get; set; } 
        public int CampusId { get; set; }
        public string TeacherName { get; set; }
        public string TeacherLastname1 { get; set; }
        public string TeacherLastname2 { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? ModifiedAt { get; set; } = DateTime.UtcNow;

        public virtual Campus Campus { get; set; } = null!;
        public virtual ICollection<Comment> Comments { get; set; }
    }
}
